internal/sqlite/sqlite.go

```diff
@@ -0,0 +1,105 @@
+package sqlite
+
+import (
+	"database/sql"
+	"fmt"
+
+	_ "modernc.org/sqlite" // import SQLite driver.
+)
+
+// DataSource wraps a SQLite DataSource.
+type DataSource struct {
+	db *sql.DB
+}
+
+// New initializes a new SQLite Datasource.
+func New(dsn string) (*DataSource, error) {
+	db, err := sql.Open("sqlite", dsn)
+	if err != nil {
+		return nil, err
+	}
+
+	return &DataSource{db: db}, nil
+}
+
+func (d *DataSource) query(query string) (data [][]*string, err error) {
+	rows, err := d.db.Query(query)
```

Comment:
You might need to close connections like here: https://github.com/KenanBek/dbui/pull/30/files#diff-75e67db3163a450c3c6d8bd333003dc643c83aece0ae3c360e732187cd419a50R93

There are two abstractions for Clossable and Commitable. Check here: https://github.com/KenanBek/dbui/blob/main/internal/dbui.go#L60-L85